### PR TITLE
 feat: [ANDROAPP-817] adds notes tab on event view

### PR DESCRIPTION
**Pull request by mmmateos**
Jira issue: https://jira.dhis2.org/browse/ANDROAPP-817
Comments: Disabled viewPager swipe on event screen

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/notes/NotesRepository.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import io.reactivex.Single
+import java.util.Date
+import org.dhis2.utils.DateUtils
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.enrollment.EnrollmentStatus
+import org.hisp.dhis.android.core.note.Note
+import org.hisp.dhis.android.core.note.NoteCreateProjection
+
+class NotesRepository(private val d2: D2, val programUid: String) {
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/notes/NotesRepository.kt]: I miss the unit test :D

File: app/src/main/java/org/dhis2/usescases/notes/NotesPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import io.reactivex.processors.FlowableProcessor
+import io.reactivex.processors.PublishProcessor
+import org.dhis2.data.schedulers.SchedulerProvider
+import org.hisp.dhis.android.core.note.Note
+import timber.log.Timber
+
+/**
+ * QUADRAM. Created by ppajuelo on 09/04/2019.
+ */
+class NotesPresenter(
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/notes/NotesPresenter.kt]: I miss the unit test :D

**Comment by JaimeToca** [comment]
Hello @mmmateos is the design of the list change and fab button removed?

File: app/src/main/java/org/dhis2/usescases/notes/NotesRepository.kt (hunk 1 above)
Comment (mmmateos) [reply app/src/main/java/org/dhis2/usescases/notes/NotesRepository.kt]: You're rigth. I forgot :sweat_smile:

File: app/src/main/java/org/dhis2/usescases/notes/NotesViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        val userName = note.storedBy()
+            ?.split(" ".toRegex())
+            ?.dropLastWhile { it.isEmpty() }
+            ?.toTypedArray()
+
+        var userInit = ""
+        userName?.forEachIndexed { index, word ->
+            if (index > 1) return
+            userInit += word.first()
+        }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/notes/NotesViewHolder.kt]: These things could be in another class a unit tested

Comment (ferdyrod) [reply app/src/main/java/org/dhis2/usescases/notes/NotesViewHolder.kt]: Better yet, we could add them as en extension function if they will be used elsewhere

**Comment by ferdyrod** [comment]
Tested on device!

The only thing I found is that is a user does not have writing permission on the program, the add note fab still appears. Should it appear or can it have different permissions on write a note?

**Comment by ferdyrod** [comment]
Tested!!

**Pull request merged**